frontend/src/components/App/Admin/AddUserModal.tsx

```diff
@@ -0,0 +1,182 @@
+import { required } from '@components/Common/Field/Validateors'
+import { APIRequestResultsRenderer } from '@components/Common/APIRequestResultsRenderer'
+
+interface Props extends RouteComponentProps<{ applicationId: string }> {
+  isModalOpen: boolean
+  acl: AccessControlList[]
+  toggle: () => void
+  reload: () => void
+}
+
+interface StateProps {
+  fetchAllUsersStatus: APIRequest<UserInfo[]>
+  saveAccessControlStatus: APIRequest<boolean>
+}
+
+interface DispatchProps {
+  addNotification: (params) => AddNotificationAction
+  fetchAllUsers: () => Promise<void>
+  saveAccessControl: (params) => Promise<void>
+}
+
+type AddUserModalProps = Props & StateProps & DispatchProps & InjectedFormProps<any, any>
+
+class AddUserModal extends React.Component<AddUserModalProps> {
+  private alreadyAdded: Set<string>
+  constructor(props) {
+    super(props)
+    this.onCancel = this.onCancel.bind(this)
+    this.onSubmit = this.onSubmit.bind(this)
+    this.renderForm = this.renderForm.bind(this)
+    this.state = {
+      submitting: false
+    }
+    this.alreadyAdded = new Set<string>()
+    props.acl.forEach((e: AccessControlList) => {
+      this.alreadyAdded.add(e.userUid)
+    })
+  }
+  componentWillReceiveProps(nextProps: AddUserModalProps) {
+    const { isModalOpen } = this.props
+    const { saveAccessControlStatus, reset, toggle, reload, submitting } = nextProps
+    if (isModalOpen && submitting) {
+      const succeeded: boolean = isAPISucceeded<boolean>(saveAccessControlStatus)
+      const failed: boolean = isAPIFailed<boolean>(saveAccessControlStatus)
+      if (succeeded) {
+        nextProps.addNotification({ color: 'success', message: 'Successfully added user' })
+        reset()
+        toggle()
+        reload()
+      }
+      if (failed) {
+        nextProps.addNotification({ color: 'error', message: 'Something went wrong. Try again later' })
+      }
+    }
+  }
+  componentWillMount() {
+    const { fetchAllUsers } = this.props
+    fetchAllUsers()
+  }
+  render() {
+    const { fetchAllUsersStatus, isModalOpen } = this.props
+    return (
+      <Modal isOpen={isModalOpen} toggle={this.onCancel}>
+        <APIRequestResultsRenderer
+          APIStatus={{ fetchAllUsersStatus }}
+          render={this.renderForm}
+        />
+      </Modal>
+    )
+  }
+  private renderForm(results) {
+    const { handleSubmit } = this.props
+    return (
+      <form onSubmit={handleSubmit(this.onSubmit)}>
+        <ModalHeader toggle={this.onCancel}>
+          <i className='fas fa-user-plus fa-fw mr-2'></i>
+          Add User
+        </ModalHeader>
+        {this.renderBodyForm(results.fetchAllUsersStatus)}
+        {this.renderFooterButtons()}
+      </form>
+    )
+  }
+  private renderBodyForm(userInfos: UserInfo[]) {
+    const roles = [
+      { label: 'view',  value: 'view'  },
+      { label: 'edit',  value: 'edit'  },
+      { label: 'admin', value: 'admin' }
+    ]
```

Comment:
These role arguments should be stored on `frontend/configs/config.json`.

Reply:
Hum.
Currently we stored the service level information (e.g. dev/stg/prod) to that file and used them when we deploy an application/service via dashboard. So I think this definition (view/edit/admin) should be stored as the same way of the service level.

Reply:
If you know the appropriate way to manage the file like this, please let us know. And if you have a time, please fix our current `config.json` of service level also.

Reply:
It seems that we put the service level information to config.json because they might be changed by dashboard users. It's right.
But in this case, permission roles (and their labels) are just constant values that aren't changed.

Reply:
Now I got it.
You are right and I totally agree with you.

> However, it seems to be better to have these values are defined elsewhere. I'll make file for constants and move them.

Please :)